app/models/miq_provision_virt_workflow.rb

```diff
@@ -760,7 +760,7 @@ def self.from_ws_ver_1_0(version, user, src_name, target_name, auto_approve, tag
     if p.validate(values) == false
       errors = []
       p.fields { |_fn, f, _dn, _d| errors << f[:error] unless f[:error].nil? }
-      raise _("Provision failed for the following reasons:\n%{errors}") % {:errors => errors.join("\n")}
+      raise Api::BadRequestError, _("Provision failed for the following reasons:\n%{errors}") % {:errors => errors.join("\n")}
```

Comment:
@abellotti @chrisarcand Should these errors be changed to "InvalidArguments" and have the API convert to a `Api::BadRequestError`?

From the request side these methods are no longer called exclusive from the API so I would prefer to not raise the Api error class here.  Maybe we will want to refactor these methods for API and non-API callers.

Reply:
Yup, you're absolutely right. API exceptions are implementation details of how the API responds to consumers and have no business in this project. You just let us know what mechanism you'd like to use here (be it raising an exception or something else) and the API should look for that and respond itself accordingly. 

If you do use an exception, which sounds likely, yes you should use a custom one to make it easier for the API to catch. 

Thanks @kruge002 for noticing and addressing this.